react/src/components/CustomizedImageList.tsx

```diff
@@ -358,93 +360,15 @@ const CustomizedImageList: React.FC = () => {
-                      return;
-                    }
-                    if (errors && errors?.length > 0) {
-                      const errorMsgList = _.map(
-                        errors,
-                        (error) => error.message,
-                      );
-                      for (const errorMsg of errorMsgList) {
-                        message.error(errorMsg);
-                      }
-                      return;
-                    }
-                    commitForget({
-                      variables: { id: row.id },
-                      onCompleted: (res, errors) => {
-                        setInFlightImageId(undefined);
-                        if (
-                          !_.isNil(res.forget_image_by_id) &&
-                          !res.forget_image_by_id.ok
-                        ) {
-                          message.error(res.forget_image_by_id.msg);
-                          return;
-                        }
-                        if (errors && errors?.length > 0) {
-                          const errorMsgList = _.map(
-                            errors,
-                            (error) => error.message,
-                          );
-                          for (const errorMsg of errorMsgList) {
-                            message.error(errorMsg);
-                          }
-                          return;
-                        }
-                        startRefetchTransition(() => {
-                          updateCustomizedImageListFetchKey();
-                        });
-                        message.success(
-                          t('environment.CustomizedImageSuccessfullyDeleted'),
-                        );
-                      },
-                      onError: () => {
-                        message.error(
-                          t('environment.FailedToDeleteCustomizedImage'),
-                        );
-                      },
-                    });
-                  },
-                  onError: () => {
-                    message.error(
-                      t('environment.FailedToDeleteCustomizedImage'),
-                    );
-                  },
-                });
-              }
+          <Button
+            type="text"
+            icon={<DeleteOutlined />}
+            onClick={() => {
+              setImageToDelete(row?.id || null);
+              setIsDeleteModalOpen(true);
             }}
-          >
-            <Button
-              type="text"
-              icon={<DeleteOutlined />}
-              danger
-              loading={
-                (isInFlightForget || isInFlightUntag) &&
-                inFlightImageId === row?.id + customizedImageListFetchKey
-              }
-              disabled={
-                (isInFlightForget || isInFlightUntag) &&
-                inFlightImageId !== row?.id + customizedImageListFetchKey
-              }
-            />
-          </Popconfirm>
+            danger
```

Comment:
You can use BAIButton with action to show loading indicator since confirm modal returns promise